src/Toolkit/Toolkit.Maui/FeatureForm/ScannerViewController.iOS.cs

```diff
@@ -0,0 +1,347 @@
+﻿#if __IOS__
+using AVFoundation;
+using UIKit;
+
+namespace Esri.ArcGISRuntime.Toolkit.Maui
+{
+    // Port of https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/81552fcff71e638a2c306ead6406a8a0c93ef334/Sources/ArcGISToolkit/Common/CodeScanner.swift
+    internal sealed class ScannerViewController : UIViewController, IAVCaptureMetadataOutputObjectsDelegate
+    {
+        private AVCaptureSession? captureSession;
+
+        private UISelectionFeedbackGenerator feedbackGenerator = new UISelectionFeedbackGenerator();
+        private AVCaptureVideoPreviewLayer? previewLayer;
+        /// The color of a code overlay before it's been targeted for auto-scan.
+        private UIColor normalOverlayColor = UIColor.White.ColorWithAlpha(0.25f);
+        /// The number of consecutive hits required to trigger an automatic scan.
+        private const int requiredTargetHits = 25;
+        private CoreAnimation.CAShapeLayer? reticleLayer;
+        /// The number of consecutive target hits. See also `requiredTargetHits`.
+        private int targetHits = 0;
+        /// The string value of the targeted code.
+        private string? targetStringValue;
+        private TaskCompletionSource<string> tcs = new TaskCompletionSource<string>();
+        private List<MetadataObjectLayer> metadataObjectOverlayLayers = new List<MetadataObjectLayer>(0);
+
+        private ScannerViewController() : base()
+        {
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+            if (disposing)
+            {
+                feedbackGenerator?.Dispose();
+                previewLayer?.Dispose();
+                normalOverlayColor?.Dispose();
+                reticleLayer?.Dispose();
+                captureSession?.Dispose();
+            }
+        }
+
+        public static Task<string> ScanAsync()
+        {
+            var scanner = new ScannerViewController();
+            var currentViewController = Microsoft.Maui.ApplicationModel.Platform.GetCurrentUIViewController();
+            currentViewController!.ShowDetailViewController(scanner, null);
+            return scanner.tcs.Task;
+        }
+        private class MetadataObjectLayer : CoreAnimation.CAShapeLayer
+        {
+
+            public AVMetadataObject? MetadataObject { get; set; }
+            public string? StringValue => (MetadataObject as AVMetadataMachineReadableCodeObject)?.StringValue;
+        }
+
+        public override void ViewDidLayoutSubviews()
+        {
+            base.ViewDidLayoutSubviews();
+            UpdateReticleAndAutoFocus();
+        }
+
+        public override async void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+            var success = await AVCaptureDevice.RequestAccessForMediaTypeAsync(AVAuthorizationMediaType.Video);
+            if (success != true)
+                return;
```

Comment:
It looks like our error-detection code just bails out and leaves the user at a blank screen.  Would you want to improve on that, e.g. telling user to grant permission or printing an error message if videoInput creation fails?

Reply:
The user will be presented with a dialog to access the camera. If they deny it, it basically just cancels out and doesn't do anything. It should be obvious why that button doesn't do anything when you already hit cancel, rather than bugging them with an additional prompt to close.